### PR TITLE
♻️refactor: amazon s3 upload functions

### PR DESCRIPTION
**Pull request by Marukome0743**
# Summary

- [x] move s3.tsx to api route
- [x] use FormData type instead of Profile type at fetch
- [x] change hidden operation at image input in order to get File

## To be Getting File object
I delete `DropImageZone` DOM when image exists previously.
However. this prevent access image file info.
So I hidden `DropImageZone` now.

## Drag and Drop image upload don't work correctly
Due to React hook form implementation,
Drag and Drop image upload function has a problem.

I think we should graduate from RHF
and reimplement with [zod](https://zod.dev/) like [Next.js documentation](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#forms).

## How to verify your codes
I checked AWS S3 file upload functions.

### CODE REVIEW

File: app/register/components/confirmDialog.tsx [hunk 1]
```diff
@@ -35,50 +29,35 @@ export function ConfirmDialog({
...
       .catch((error) => {
-        alert(error)
-        return
+        alert(`画像アップロードエラー：${error}`)
       })
+
+    fetch('/api/user', {
+      method: 'POST',
+      body: formData,
+    })
```
Comment (kazutan1230) [inline app/register/components/confirmDialog.tsx:51]: It looks like lengthly to post the _formData_ twice.
We may should post the _formData_ to one API,
extract the image file from _formData_,
then send it to the S3 by POST,
and send user profile to the database when POST image success,
or send error back to client when POST fail.

File: app/api/image/route.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const body = await request.formData()
+  const image = body.get('image') as File
+  const arrayBuffer = await image.arrayBuffer()
+  const buffer = Buffer.from(arrayBuffer)
+
+  const command = new PutObjectCommand({
+    ACL: 'private',
+    Bucket: process.env.S3_BUCKET,
+    ContentType: image.type,
+    Key: image.name,
```
Comment (kazutan1230) [inline app/api/image/route.ts:24]: Maybe we use this key when download the file, don't we?
Present condition, the users who upload the file can name it.
I'm afraid of uploaded same _image.name_ files although that possibility is low.
In that case, can the download users get the file correctly?

File: app/lib/s3.tsx
Comment (kazutan1230) [inline app/lib/s3.tsx:1]: As we talked yesterday, this file is modified and remains.

File: app/register/components/confirmDialog.tsx (hunk 1 above)
Comment (kazutan1230) [inline app/register/components/confirmDialog.tsx:51]: It looks like lengthly to post the formData twice.
We may should post the formData to one API,
extract the image file from formData,
then send it to the S3 by POST,
and send user profile to the database when POST image success,
or send error back to client when POST fail.

File: app/register/components/confirmDialog.tsx [hunk 3]
```diff
@@ -35,50 +29,35 @@ export function ConfirmDialog({
...
-          // values.image[0] からファイル取得困難のためdbdataにs3に保存したファイル名を押し込む。
-          const dbdata = Object.assign(values, { fileName: fileName })
-          fetch('/api/user', {
-            method: 'POST',
-            headers: { 'Content-Type': 'application/json' },
-            body: JSON.stringify(dbdata),
-          })
-            .then((res) => {
-              if (!res.ok) {
-                // DB登録失敗時の画像削除処理は未実装
```
Comment (kazutan1230) [inline app/register/components/confirmDialog.tsx:67]: It looks this PR doesn't resolve the problem mentioned in this comment,
so I think we shouldn't remain the comment or should make issue.

File: app/register/components/confirmDialog.tsx [hunk 4]
```diff
@@ -35,50 +29,35 @@ export function ConfirmDialog({
...
-                alert(res.statusText)
-                return
-              }
-              router.push('/register/success')
-            })
-            .catch((error) => {
-              // DB登録失敗時の画像削除処理は未実装
```
Comment (kazutan1230) [inline app/register/components/confirmDialog.tsx:74]: It looks this PR doesn't resolve the problem mentioned in this comment,
so I think we shouldn't remain the comment or should make issue.

File: app/register/components/confirmDialog.tsx [hunk 5]
```diff
@@ -35,50 +29,35 @@ export function ConfirmDialog({
...
-              alert(error)
-            })
-        } else {
-          // DB登録失敗時の画像削除処理は未実装
```
Comment (kazutan1230) [inline app/register/components/confirmDialog.tsx:78]: It looks this PR doesn't resolve the problem mentioned in this comment,
so I think we shouldn't remain the comment or should make issue.

Comment (Marukome0743) [reply app/register/components/confirmDialog.tsx:78]: Fetch requests don't save image data if fetch failed.
That's why you don't need to delete image.

File: app/register/components/confirmDialog.tsx (hunk 4 above)
Comment (Marukome0743) [reply app/register/components/confirmDialog.tsx:74]: Fetch requests don't save image data if fetch failed.
That's why you don't need to delete image.

File: app/register/components/confirmDialog.tsx (hunk 3 above)
Comment (Marukome0743) [reply app/register/components/confirmDialog.tsx:67]: Fetch requests don't save image data if fetch failed.
That's why you don't need to delete image.

Comment (kazutan1230) [reply app/register/components/confirmDialog.tsx:67]: There will be two fetch requests, image data fetch to the S3 and user profile fetch to the database.
If the S3 fetch failed, fetch requests don't save image data.
When the S3 fetch successes and the database fetch fails,
image data is saved in the S3 and no data is saved in the database.
So, I think I need to delete image.
Same as below comment, so I'll resolve conversation there.

File: app/register/components/confirmDialog.tsx (hunk 1 above)
Comment (Marukome0743) [reply app/register/components/confirmDialog.tsx]: I changed the fetch call time at once.

File: app/api/image/route.ts (hunk 2 above)
Comment (Marukome0743) [reply app/api/image/route.ts]: I add employeeId to Key value to be unique.

File: app/register/components/confirmDialog.tsx (hunk 3 above)
Comment (Marukome0743) [reply app/register/components/confirmDialog.tsx:67]: OK, I unify a fetch function.

You should adding the delete function to api route
if you have to delete image data.

So comment area is not fetch function, is api route.

**Review by kazutan1230 — APPROVED** [review]
I approve this pr.

**Pull request merged**